src/platform/unix/mod.rs

```diff
@@ -593,13 +598,15 @@ impl OsIpcOneShotServer {
         unsafe {
             let fd = libc::socket(libc::AF_UNIX, SOCK_SEQPACKET, 0);
             let mut path: Vec<u8>;
+            let dir = TempDir::new().unwrap();
```

Comment:
Please use `tempfile::Builder` (https://docs.rs/tempfile/3.0.2/tempfile/struct.Builder.html), to get a meaningful directory name. (The actual file name can then be shortened to just "socket" I guess.)